engine/src/test/java/org/hibernate/validator/test/internal/engine/cascaded/NestedCascadedConstraintsTest.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.test.internal.engine.cascaded;
+
+import static org.hibernate.validator.testutil.ConstraintViolationAssert.assertCorrectConstraintTypes;
+import static org.hibernate.validator.testutil.ConstraintViolationAssert.assertCorrectPropertyPaths;
+import static org.hibernate.validator.testutil.ConstraintViolationAssert.assertNumberOfViolations;
+import static org.hibernate.validator.testutils.ValidatorUtil.getValidator;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import javax.validation.ConstraintViolation;
+import javax.validation.Valid;
+import javax.validation.Validation;
+import javax.validation.Validator;
+import javax.validation.constraints.NotNull;
+
+import org.hibernate.validator.HibernateValidator;
+import org.hibernate.validator.constraints.Email;
+import org.hibernate.validator.testutil.TestForIssue;
+import org.testng.annotations.Test;
+
+@TestForIssue(jiraKey = "HV-1237")
+public class NestedCascadedConstraintsTest {
+
+	@Test
+	public void testNestedValid() {
+		Validator validator = getValidator();
+
+		Set<ConstraintViolation<EmailAddressMap>> constraintViolations = validator.validate( EmailAddressMap.validEmailAddressMap() );
+
+		assertNumberOfViolations( constraintViolations, 0 );
+
+		constraintViolations = validator.validate( EmailAddressMap.invalidEmailAddressMap() );
+
+		assertNumberOfViolations( constraintViolations, 2 );
+		assertCorrectPropertyPaths(
+				constraintViolations,
+				"map[invalid].[1].email",
```

Comment:
@gsmet if you check out http://staging.beanvalidation.org/latest-draft/spec/#validationapi-constraintviolation, note that key/index/inIterabler are only set on the following node. This may be the cause of confusion.

So I think I'd expect these three nodes here (not totally sure about the kind and name of the 2nd node, but that seems sensible):

    node( kind = PROPERTY, name = "map", key = null, index = null, inIterable = false )
    node( kind = TYPE_ARGUMENT, name = "<map value>", key = "invalid", index = null, inIterable = true )
    node( kind = PROPERTY, name = "email", key = null, index = 1, inIterable = true )